adapters/clients/client.go

```diff
@@ -85,6 +88,12 @@ func (r *retryer) retry(ctx context.Context, n int, work func(context.Context) (
 		if delay = backOff(delay); delay > r.maxBackOff {
 			delay = r.maxBackOff
 		}
+		if retryLogEntry != nil {
+			retryLogEntry.WithFields(logrus.Fields{
+				"numberOfRetriesLeft": n,
+				"delay":               delay,
+			}).WithError(err).Debug("retrying after work errored")
```

Comment:
I have all the new logs as debug logs because i think these could be quite spammy as info logs, but on the other hand i assume most users have a log level or info or warn, so if our logs were cleaner than they currently are I would lean towards making these info or warn. Would love to hear others' thoughts.

Reply:
Maybe I could switch to the warn level if there are > N retries? Would take some refactoring and might be overkill. I also need to verify that we see useful error logs when we run out of retries or time out.

Reply:
Totally depends on how actionable these logs are IMHO.

Reply:
IMO we should have them as debug as kavi said they are not really actionnable and only an info for debugging issues.

We can log *once* as info that a request is slow, or that a bunch of requests are being retried internally. You can't really use the `URL` to detect uniqueness was these will be almost unique for class/shard (in an MT setup at least, less so in single tenant setup).

Maybe make it so that at "worst" it will print the log line as info every X minutes/seconds ? This way if 10k RPS are being heavily retried we end up with just a few warn logs that tells us "X requests were retried internally more than Y times in the last Z minutes/seconds" ?

Reply:
Yea sounds good to me. Keep an eye for noisy logs since it's inside a loop. I'm happy to go as is 👍